src/server/master_server/chunk_server_manager.cc

```diff
@@ -1,69 +1,304 @@
 #include "src/server/master_server/chunk_server_manager.h"
 
+namespace protos {
+bool operator==(const protos::ChunkServerLocation& lhs,
+                const protos::ChunkServerLocation& rhs) {
+  return lhs.server_hostname() == rhs.server_hostname() &&
+         lhs.server_port() == rhs.server_port();
+}
+
+bool operator==(const protos::ChunkServer& lhs,
+                const protos::ChunkServer& rhs) {
+  return lhs.location() == rhs.location();
+}
+}  // namespace protos
+
 namespace gfs {
 
 namespace server {
 
-// Iterates through the chunk servers and selects the server with
```

Comment:
I am wondering whether we could simplify things a bit here by just randomly select a number of chunk servers? given that we are not having a super sophisticated design. Provided all chunk server machines are the same (in our case), randomly selection seems tolerable?

Reply:
Correct, but the problem with random selection is you have to do periodic load balancing and end up moving chunks from one server to another. So this helps to avoid that by doing balanced distribution during allocation.